test/e2e/autoscaling_utils.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+)
+
+//		ConsumingRC is a tool for testing. It helps create specified usage of CPU or memory (Warnig: memory not supported)
+// 		typical use case:
+// 		rc.ConsumeCPU(600)
+//		// ... check your assumption here
+//		rc.ConsumeCPU(300)
+//		// ... check your assumption here
+type ConsumingRC struct {
+	name          string
+	milicores     int
+	mutex         *sync.Mutex
+	f             *Framework
+	stopConsuming bool
+}
+
+// creates new ConsumingRC
+func NewConsumingRC(name string, replicas int, f *Framework) *ConsumingRC {
+	startService(f.Client, f.Namespace.Name, name, replicas)
+	rc := &ConsumingRC{
+		name:          name,
+		milicores:     0,
+		mutex:         &sync.Mutex{},
+		f:             f,
+		stopConsuming: false,
+	}
+	go rc.consumeCPU()
+	return rc
+}
+
+// consumes given number of CPU
+func (rc *ConsumingRC) ConsumeCPU(milicores int) {
+	rc.mutex.Lock()
+	rc.milicores = milicores
+	rc.mutex.Unlock()
+}
+
+func (rc *ConsumingRC) consumeCPU() {
+	defer GinkgoRecover()
+	durationSec := 10
+	for {
+		rc.mutex.Lock()
+		if rc.stopConsuming {
+			return
+		}
+		count := rc.milicores / 100
+		rest := rc.milicores - count*100
+		rc.sendConsumeCPUrequests(count, 100, durationSec)
```

Comment:
Usually it is not the best idea to do network communication under a lock. Is it necessary?